### PR TITLE
[SPARK-20646][core] Port executors page to new UI backend.

### PR DESCRIPTION
**Pull request by vanzin**
The executors page is built on top of the REST API, so the page itself
was easy to hook up to the new code.

Some other pages depend on the `ExecutorListener` class that is being
removed, though, so they needed to be modified to use data from the
new store. Fortunately, all they seemed to need is the map of executor
logs, so that was somewhat easy too.

The executor timeline graph required adding some properties to the
ExecutorSummary API type. Instead of following the previous code,
which stored all the listener events in memory, the timeline is
now created based on the data available from the API.

I had to change some of the test golden files because the old code would
return executors in "random" order (since it used a mutable Map instead
of something that returns a sorted list), and the new code returns executors
in id order.

Tested with existing unit tests.

### CODE REVIEW

**Comment by vanzin** [comment]
For context:

* Project link: https://issues.apache.org/jira/browse/SPARK-18085
* Upcoming PRs that build on this code: https://github.com/vanzin/spark/pulls

File: core/src/main/scala/org/apache/spark/ui/exec/ExecutorsTab.scala [hunk 1]
```diff
@@ -17,192 +17,46 @@
...
 package org.apache.spark.ui.exec
 
-import scala.collection.mutable.{LinkedHashMap, ListBuffer}
+import javax.servlet.http.HttpServletRequest
+
+import scala.xml.Node
 
 import org.apache.spark.{Resubmitted, SparkConf, SparkContext}
-import org.apache.spark.annotation.DeveloperApi
 import org.apache.spark.scheduler._
```
Comment (squito) [inline core/src/main/scala/org/apache/spark/ui/exec/ExecutorsTab.scala]: nit: I think more of these imports can be deleted

File: core/src/test/scala/org/apache/spark/ui/StagePageSuite.scala [hunk 2]
```diff
@@ -22,13 +22,14 @@ import javax.servlet.http.HttpServletRequest
...
 import scala.xml.Node
 
+import org.mockito.Matchers.anyString
 import org.mockito.Mockito.{mock, when, RETURNS_SMART_NULLS}
 
 import org.apache.spark._
 import org.apache.spark.executor.TaskMetrics
 import org.apache.spark.scheduler._
+import org.apache.spark.status.AppStatusStore
 import org.apache.spark.storage.StorageStatusListener
```
Comment (squito) [inline core/src/test/scala/org/apache/spark/ui/StagePageSuite.scala]: nit: this import is no longer used

**Review by squito — APPROVED** [review]
lgtm

a couple nits about unused imports -- there are some more in AppStatusStore, SparkUI as well (probably from before) would be nice to clean them up.

**Comment by squito** [comment]
Also going to copy this comment from here for others: https://github.com/vanzin/spark/pull/45

>> squito 12 days ago
>> what do you think about adding a helper method for these json snippets? obviously not the fault of your change, but I find these so hard to read, eg. figuring out where data-title ends (worse in some of the other cases esp in StagePage). something like

```
UIUtils.timelineJsonEvent(
  className="executor added",
  group="executors",
  start = e.addTime.getTime(),
  end = None,
  contentClass = "executor-event-content",
  dataTitle = s"Executor ${e.id}<br> Added at ${UIUtils.formatDate(new Date(e.addTime.getTime()))}",
  contentBody = s"Executor ${e.id} added")
```

> vanzin 11 days ago Owner
> I'm not against it but I'd rather do that separately.

agree, fine to do that later.

**Comment by squito** [comment]
lgtm
(assuming tests pass)

**Comment by vanzin** [comment]
retest this please

**Comment by squito** [comment]
merged to master

**Pull request closed without merging**